src/huntsman/pocs/utils/huntsman.py

```diff
@@ -63,6 +64,10 @@ def create_huntsman_scheduler(observer=None, logger=None, *args, **kwargs):
                            HuntsmanMoonAvoidance(),
                            SunAvoidance()]
 
+            # add global observe each target once constraint
+            if get_config('constraints.observe_once', default=False):
+                constraints.append(AlreadyVisited())
```

Comment:
The scheduler will veto items in order, so it's best if the constraints can veto early in the process. If it's already been visited there is no reason to check, e.g. distance from the moon.

```suggestion
                constraints.insert(0, AlreadyVisited())
```